examples/gno.land/p/demo/dao/proposals.gno

```diff
@@ -58,5 +61,5 @@ type Proposal interface {
 	IsExpired() bool
 
 	// Render renders the proposal in a readable format
-	Render() string
+	Render(idx int, authorUsername string) string
```

Comment:
The caller shouldn't need to specify these things, the `Proposal` should know about them internally

Reply:
It can be a phishing vector

Reply:
Two issues: 

- The `Proposal` interface doesn't have a built-in `ID` - should we add this? Otherwise, prop page rendering can be done inside of the `r/gov/v2` realm, but it will be split up between `gov/v2` & `simpledao`, making it a bit ugly.
- This is a `p/` - to resolve usernames, we need to access `r/demo/users`; importing a `r/` inside of a `p/` will be forbidden soon. We can set the username upon prop creation, but it will not be updated later if a user is not registered initially and then registers.

Reply:
I did this; check it out:

[05253fc](https://github.com/gnolang/gno/pull/3096/commits/05253fc5425412207ab4a0b025cf4337f4448ba8) & [6ed08b0](https://github.com/gnolang/gno/pull/3096/commits/6ed08b0677237696b863a8323464919b6aa6cf97)